### PR TITLE
Re-enable priority test cases in Microsoft.Data.Web.UnitTests suite

### PR DESCRIPTION
**Pull request by AlanWong-MS**
### Description
This change modifies and re-enables relevant tests that had been skipped in the Microsoft.Data.Web.UnitTests test suite. These tests help fill important missing test gaps for our code coverage. A separate pull request will be sent out to address the rest of the skipped tests.

### Checklist (Uncheck if it is not completed)
- [ x ] Test cases added
- [ x ] Build and test with one-click build and test script passed

### CODE REVIEW

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 1]
```diff
@@ -951,10 +951,18 @@ public static IEdmEntitySet FindDeclaredEntitySet(this IEdmModel model, string q
...
-                if (container != null)
+                try
+                {
+                    IEdmEntityContainer container = model.EntityContainer;
+                    if (container != null)
+                    {
+                        return container.FindEntitySetExtended(qualifiedName);
+                    }
+                }
+                catch (NotImplementedException)
```
Comment (AlanWong-MS) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: If we perform any logging, I think it'd be appropriate to track any exceptions here.

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 2]
```diff
@@ -951,10 +951,18 @@ public static IEdmEntitySet FindDeclaredEntitySet(this IEdmModel model, string q
...
                 {
-                    return container.FindEntitySetExtended(qualifiedName);
+                    // model.EntityContainer can throw NotImplementedException
+                    return foundEntitySet;
```
Comment (robward-ms) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >foundEntitySet [](start = 27, length = 14)

Is this null in this case? Would it make more sense to be explicit and return null?  Same below.

**Comment by robward-ms** [comment]
>         public static XmlDocument VerifyXPaths(Stream resultStream, string responseFormat, string[] web3sXPaths, string[] jsonXPaths, string[] atomXPaths)

This variable is not used?

---
Refers to: test/FunctionalTests/Tests/DataServices/UnitTests/ServerUnitTests/Utils/UnitTestsUtil.cs:258 in a57c94d. [](commit_id = a57c94dcb47125db8f1ca4758eba154c4f0c19aa, deletion_comment = False)

**Comment by robward-ms** [comment]
:clock1:

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 2 above)
Comment (AlanWong-MS) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: Yes, that's a good idea. I'll change for the case below as well.

**Comment by mtroth-microsoft** [comment]
:shipit:

**Pull request merged**